### ISSUE TITLE
boto_secgroup forbids valid IP protocols

### ISSUE THREAD

**Issue by kraney** [saltstack/salt#28972]
boto_secgroup insists on IP protocol values from the list 'tcp', 'udp', 'icmp', 'all', '-1', however AWS accepts custom protocols. For example IP protocol '47' represents GRE traffic. In the AWS GUI, you can type 'GRE' and it's accepted, but converted to the numeric 47. It would probably be sufficient in boto_secgroup to accept numbers. E.g.

``` python
        supported_protocols = ['tcp', 'udp', 'icmp', 'all', '-1']
        if ip_protocol not in supported_protocols and not ip_protocol.isdigit():
            msg = ('Invalid ip_protocol {0} specified in security group rule.')
            raise SaltInvocationError(msg.format(ip_protocol))
```

**Comment by ryan-lane** [comment]
Ah. Interesting. I didn't know that. This check is mostly here as a ways to avoid outages, due to the way the API works. We can modify this to accept the named protocols or any number. Ideally if there's a way to know the range of valid numbers it would be nice to check it.

The issue with the API is that for secgroup rules, it requires you to delete a rule, then add a rule. You can't update an existing rule. So, if you update a rule and specify an invalid protocol, it'll just remove your rule. That behavior is really dangerous, so we try to protect against it as much as possible.

**Comment by lorengordon** [comment]
@ryan-lane, 0-255 should be the range of valid integers for IP protocols.

http://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml

**Comment by Ch3LL** [comment]
Just as a headsup this was fixed in #28980 but there was an issue found here: #36961 related to this change. Is everyone okay closing this here and we will monitor this new issue there?

**Comment by rallytime** [comment]
I have fixed the issue raised in #36961 with #37447. Since we haven't heard back about the original request above, I'll close this up.

If this pops up again, let us know and we can re-open. Thanks!
